lld/ELF/Arch/LoongArch.cpp

```diff
@@ -782,6 +886,10 @@ static bool relax(Ctx &ctx, InputSection &sec) {
       }
       break;
     }
+    case R_LARCH_PCALA_HI20:
+      if (isPair(relocs, i) && relocs[i + 2].type == R_LARCH_PCALA_LO12)
```

Comment:
i+2 causes overflow?

Reply:
> i+2 causes overflow?

isPair will check this.

Reply:
> ```
> R_LARCH_PCALA_HI20   <-   i
> R_LARCH_RELAX             <- i + 1 == reloc.size() - 1
> END
> 
> Check relaxable(reloc, i) is OK.
> When check relaxable(reloc, i + 2), due to the "i + 2 + 1" != reloc.size() is true, may it overflow?
> ```

It seems that in the isPair function, it would be better to change i + 1 ! = relocs.size() to i + 1 < relocs.size() .